core/src/main/java/org/apache/calcite/rel/rules/AggregateValuesRule.java

```diff
@@ -75,10 +91,18 @@ public AggregateValuesRule(RelBuilderFactory relBuilderFactory) {
   @Override public void onMatch(RelOptRuleCall call) {
     final Aggregate aggregate = call.rel(0);
     final Values values = call.rel(1);
-    Util.discard(values);
     final RelBuilder relBuilder = call.builder();
     final RexBuilder rexBuilder = relBuilder.getRexBuilder();
 
+    if ((aggregate.getGroupCount() != 0 || !values.getTuples().isEmpty())
+        && aggregate.groupSets.size() == 1) {
+      List<ImmutableList<RexLiteral>> distinctValues =
+          values.getTuples().stream().distinct().collect(Collectors.toList());
```

Comment:
this distinct uses Java equals(). I assume it is always correct, but I am not sure it will always find all duplicates.

Reply:
Your are right, but IMO maybe can handle with it by this way, due to RelBuilder##values can only handle RexLiteral type, and all type of RexLiteral had implement Compare interface which may remove duplicates correctly.